packages/paste-core/components/slider/src/Slider.tsx

```diff
@@ -0,0 +1,142 @@
+import {useSliderState, useSlider, useSliderThumb} from '@twilio-paste/react-spectrum-library';
+
+import {SliderThumb} from './SliderThumb';
+import {StyledTrack} from './StyledTrack';
+
+export interface SliderProps {
+  element?: BoxProps['element'];
+  id: string;
+  disabled?: boolean;
+  hasError?: boolean;
+  hideRangeLabels?: boolean;
+  minValue?: number;
+  maxValue?: number;
+  step?: number;
+  /**
+   * Used to adjust how the numbers are rendered and interpreted.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat \
+   */
+  numberFormatter: Intl.NumberFormat;
+  value?: number;
+  onChange?: (value: number) => void;
+  /** Fired when the slider stops moving, due to being let go. */
+  onChangeEnd?: (value: number) => void;
+}
+
+export const Slider: React.FC<SliderProps> = (props) => {
+  const inputRef = React.useRef(null);
+  const trackRef = React.useRef(null);
+  const [hovered, setHovered] = React.useState(false);
+  const [focused, setFocused] = React.useState(false);
+
+  /*
+   * We deconstruct the props here to get the values we need
+   * and to preserve the underlying props object variable
+   */
+  const {
+    element = 'SLIDER',
+    minValue = 0,
+    maxValue = 100,
+    hideRangeLabels = false,
+    disabled: isDisabled = false,
+    id,
+  } = props;
+
+  // Our API differs from the underlying library, so we need to remap props
+  const remappedProps = {
+    ...props,
+    id,
+    isDisabled,
+    // needed to silence react-aria a11y guardrails
+    'aria-labelledby': id,
+  };
+
+  // These hooks manage the state of the slider
+  const state = useSliderState(remappedProps);
+  const {trackProps} = useSlider(remappedProps, state, trackRef);
+  const {thumbProps, inputProps, isDragging} = useSliderThumb(
+    {
+      index: 0,
+      trackRef,
+      inputRef,
+    },
+    state
+  );
+
+  // Used to determine the UI styling of the track and thumb
+  const uiStateProps = React.useMemo(() => {
+    return {
+      disabled: isDisabled,
+      error: props.hasError,
+      hovered,
+      focused,
+      dragging: isDragging,
+    };
+  }, [isDisabled, props.hasError, hovered, focused, isDragging]);
+
+  return (
+    <Box role="group" element={element}>
+      {/* Create a container for the optional min and max values */}
+      {!hideRangeLabels && (
+        <Box
+          role="presentation"
+          element={`${element}_RANGE_LABELS`}
+          display="flex"
+          justifyContent="space-between"
+          fontSize="fontSize30"
+          lineHeight="lineHeight30"
+          fontWeight="fontWeightSemibold"
+          color="colorTextWeak"
+        >
+          <Box as="span" element={`${element}_RANGE_LABELS_MIN`}>
+            {props.numberFormatter.format(minValue)}
+          </Box>
+          <Box as="span" element={`${element}_RANGE_LABELS_MAX`}>
+            {props.numberFormatter.format(maxValue)}
+          </Box>
```

Comment:
suggestion: this is a tricky one. Right now voice over is treating the "presentation" element as the parent or group. When I navigate to it what happens is I only get one element, which announces both the min and the max value in one go. I think it would be better if they were navigable separately (maybe its because they are `span` and it treats them as inline text), along with a little screen reader text desrcibing that one is the min and one is the max values. Ideally the min and max attributes on the input itself would do the job, then you could just hide this entirely, but it doesn't do anything in VO at least.